File: app/src/main/java/com/aero51/moviedatabase/ui/EpgAllProgramsFragment.java
package com.aero51.moviedatabase.ui;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.aero51.moviedatabase.R;
import com.aero51.moviedatabase.repository.model.epg.ChannelWithPrograms;
import com.aero51.moviedatabase.ui.adapter.EpgAllProgramsAdapter;
import com.aero51.moviedatabase.utils.Constants;
import com.aero51.moviedatabase.viewmodel.SharedViewModel;


public class EpgAllProgramsFragment extends Fragment {



    private SharedViewModel sharedViewModel;
    private RecyclerView recycler_view_all_programs;

    public EpgAllProgramsFragment() {
        // Required empty public constructor
    }



    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        sharedViewModel = new ViewModelProvider(getActivity()).get(SharedViewModel.class);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_epg_all_programs, container, false);
        recycler_view_all_programs = view.findViewById(R.id.all_programs_recycler_view);
        recycler_view_all_programs.setHasFixedSize(true);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext());
        recycler_view_all_programs.setLayoutManager(linearLayoutManager);
        recycler_view_all_programs.addItemDecoration(new DividerItemDecoration(recycler_view_all_programs.getContext(), linearLayoutManager.getOrientation()));

        sharedViewModel.getLiveDataChannelWithPrograms().observe(getViewLifecycleOwner(), new Observer<ChannelWithPrograms>() {
            @Override
            public void onChanged(ChannelWithPrograms channelWithPrograms) {

                EpgAllProgramsAdapter adapter =new EpgAllProgramsAdapter(channelWithPrograms);
                recycler_view_all_programs.setAdapter(adapter);
                recycler_view_all_programs.scrollToPosition(channelWithPrograms.getNearestTimePosition());
            }
        });
        Toolbar toolbar = (Toolbar) getActivity().findViewById(R.id.toolbar);
        //toolbar.setTitle("text");
        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                getActivity().onBackPressed();
                Log.d(Constants.LOG, "Toolbar clicked!");
                showBackButton(false);
            }
        });
        showBackButton(true);
        return view;
    }
    public void showBackButton(boolean show) {
        if (getActivity() instanceof AppCompatActivity) {
            ((AppCompatActivity) getActivity()).getSupportActionBar().setDisplayHomeAsUpEnabled(show);
        }
    }
}

File: app/src/main/java/com/aero51/moviedatabase/ui/EpgAllProgramsFragment.kt
package com.aero51.moviedatabase.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.aero51.moviedatabase.R
import com.aero51.moviedatabase.databinding.FragmentEpgAllProgramsBinding
import com.aero51.moviedatabase.ui.adapter.EpgAllProgramsAdapter
import com.aero51.moviedatabase.viewmodel.SharedViewModel

class EpgAllProgramsFragment : Fragment() {
    private var binding: FragmentEpgAllProgramsBinding? = null
    private var sharedViewModel: SharedViewModel? = null
    private val recycler_view_all_programs: RecyclerView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        sharedViewModel = ViewModelProvider(requireActivity()).get(SharedViewModel::class.java)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        binding = FragmentEpgAllProgramsBinding.inflate(inflater, container, false)
        binding!!.allProgramsRecyclerView.setHasFixedSize(true)
        val linearLayoutManager = LinearLayoutManager(context)
        binding!!.allProgramsRecyclerView.layoutManager = linearLayoutManager
        binding!!.allProgramsRecyclerView.addItemDecoration(DividerItemDecoration(context, linearLayoutManager.orientation))
        sharedViewModel!!.liveDataChannelWithPrograms.observe(viewLifecycleOwner, { channelWithPrograms ->
            val adapter = EpgAllProgramsAdapter(channelWithPrograms)
            binding!!.allProgramsRecyclerView.adapter = adapter
            binding!!.allProgramsRecyclerView.scrollToPosition(channelWithPrograms.nearestTimePosition!!)
        })
        val toolbar = requireActivity().findViewById<View>(R.id.toolbar) as Toolbar
        //toolbar.setTitle("text");
        toolbar.setNavigationOnClickListener {
            requireActivity().onBackPressed()
            showBackButton(false)
        }
        showBackButton(true)
        return binding!!.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }

    fun showBackButton(show: Boolean) {
        if (activity is AppCompatActivity) {
            (activity as AppCompatActivity?)!!.supportActionBar!!.setDisplayHomeAsUpEnabled(show)
        }
    }
}